src/consensus/parlia/vote/journal.rs

```diff
@@ -0,0 +1,36 @@
+use super::*;
+use hashlink::LruCache;
+use parking_lot::Mutex;
+use std::sync::Arc;
+
+const MAX_CACHE_RECENT_SIZE: usize = 512;
+
+pub struct VoteJournal {
+    buffer: Mutex<LruCache<u64, VoteEnvelope>>,
+}
+
+impl VoteJournal {
```

Comment:
WAL is required, otherwise all the votes will be lost after the node restart, need to find a 3rd package to do WAL persist